protostar/commands/test/messages/testing_summary_message.py

```diff
@@ -0,0 +1,240 @@
+        )
+    if passed_count > 0:
+        test_suites_result.append(
+            log_color_provider.colorize("GREEN", f"{passed_count} passed")
+        )
+    if total_count > 0:
+        test_suites_result.append(f"{total_count} total")
+
+    return test_suites_result
+
+
+def _get_test_suites_summary(
+    testing_summary: TestingSummary, collected_test_suites_count: int
+) -> str:
+    passed_test_suites_count = 0
+    failed_test_suites_count = 0
+    broken_test_suites_count = 0
+    total_test_suites_count = len(testing_summary.test_suites_mapping)
+    for suit_case_results in testing_summary.test_suites_mapping.values():
+        partial_summary = TestingSummary(
+            suit_case_results, testing_summary.testing_seed
+        )
+
+        if len(partial_summary.broken_suites) + len(partial_summary.broken) > 0:
+            broken_test_suites_count += 1
+            continue
+
+        if len(partial_summary.failed) > 0:
+            failed_test_suites_count += 1
+            continue
+
+        if len(partial_summary.passed) > 0:
+            passed_test_suites_count += 1
+
+    test_suites_result: list[str] = []
+
+    if broken_test_suites_count > 0:
+        test_suites_result.append(
+            log_color_provider.colorize("RED", f"{broken_test_suites_count} broken")
+        )
+    if failed_test_suites_count > 0:
+        test_suites_result.append(
+            log_color_provider.colorize("RED", f"{failed_test_suites_count} failed")
+        )
+    if passed_test_suites_count > 0:
+        test_suites_result.append(
+            log_color_provider.colorize("GREEN", f"{passed_test_suites_count} passed")
+        )
+    if total_test_suites_count > 0:
+        test_suites_result.append(f"{total_test_suites_count} total")
+
+    return ", ".join(
+        _get_preprocessed_core_testing_summary(
+            broken_count=broken_test_suites_count,
+            failed_count=failed_test_suites_count,
+            passed_count=passed_test_suites_count,
+            total_count=collected_test_suites_count,
+        )
+    )
+
+
+def _get_test_cases_summary(
+    testing_summary: TestingSummary, collected_test_cases_count: int
+) -> str:
+    broken_test_cases_count = len(testing_summary.broken)
+    failed_test_cases_count = len(testing_summary.failed)
+    passed_test_cases_count = len(testing_summary.passed)
+
+    return ", ".join(
+        _get_preprocessed_core_testing_summary(
+            broken_count=broken_test_cases_count,
+            failed_count=failed_test_cases_count,
+            passed_count=passed_test_cases_count,
+            total_count=collected_test_cases_count,
+        )
+    )
+
+
+@dataclass
+class TestingSummaryResultMessage(StructuredMessage):
+    test_collector_result: TestCollector.Result
+    testing_summary: TestingSummary
+    slowest_tests_to_report_count: int
+
+    _bold_begin = "\033[1m"
+    _bold_end = "\033[0m"
```

Comment:
Why not use `fmt.bold`?

Reply:
[use fmt.bold](https://github.com/software-mansion/protostar/pull/1254/commits/16641b396eb7a339c8849ab928b053e4fd4207f1)